lsp-ui-imenu.el

```diff
@@ -58,12 +59,27 @@
 (defcustom lsp-ui-imenu-colors '("deep sky blue" "green3")
   "Color list to cycle through for entry groups."
   :type '(repeat color)
-  :group 'lsp-ui-menu)
+  :group 'lsp-ui-imenu)
 
 (defcustom lsp-ui-imenu-window-width 0
   "When not 0, don't fit window to buffer and use value as window-width."
   :type 'number
-  :group 'lsp-ui-menu)
+  :group 'lsp-ui-imenu)
+
+(defcustom lsp-ui-imenu-auto-refresh t
+  "Automatically refresh imenu when certain conditions meet."
```

Comment:
Please set this to `nil` by default, as discussed this shouldn't be on by default. Until we make the request async.
And in the docstring please mention that it might slow down navigation/editing because the request is not async.

Reply:
This is for `after-save`. I have temporary created another variable `lsp-ui-imenu-refresh-after-change` to disable automatically refresh imenu when `after-change`. So by default; the user can still have auto refresh when `after-save`, but not for the `after-change` because the variable `lsp-ui-imenu-refresh-after-change` is already set to `nil` by default. Let me know if you still want this to be disable by default.

Reply:
hmm what about having just one variable `lsp-ui-imenu-auto-refresh` with 3 values:
- `t` for always
- `after-save` to ignore `after-change-functions`
- `nil` to disable

This will avoid having another variable l`sp-ui-imenu-refresh-after-change`

In all cases, both behaviors should be disabled by default, until we make it async

Reply:
> hmm what about having just one variable lsp-ui-imenu-auto-refresh with 3 values:

Changed.

> In all cases, both behaviors should be disabled by default, until we make it async

Got you! I have make the variable default to `nil`.